python-frontend/src/main/java/org/sonar/python/semantic/SymbolUtils.java

```diff
@@ -252,6 +251,42 @@ public static Optional<FunctionSymbol> getOverriddenMethod(FunctionSymbol functi
     return Optional.empty();
   }
 
+  public static Optional<FunctionSymbol> getFunctionSymbol(@Nullable Symbol symbol) {
```

Comment:
While I agree with the idea of having this as common logic, I'm not sure this is the best approach.

In rules such as S2638 where we care about the number and names of parameters, it makes sense to consider ambiguous symbols alternatives as equivalent when their parameters have same number and names.

However, for some other rules, we might care about the type hint of the parameters, or the return type, in which case this method might not behave as expected.

Probably we could have some disambiguation strategies that are clearly defined by the caller (either by having multiple methods, some argument or some predicate) that would return a single function / class symbol from an ambiguous symbol only if the criteria are met. And maybe we don't need to do it in this PR and have a separate one for that (or even a ticket). Happy to discuss this.